tests/edges/test_bsplinecurve.py

```diff
@@ -17,6 +17,66 @@
 folder = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'bsplinecurve_objects')
 
 
+class TestBSplineCurve(unittest.TestCase):
+    def test_arc_to_nurbs(self):
+        arc = vme.Arc2D.from_3_points(volmdlr.Point2D(0, 0.3), volmdlr.Point2D(1, -0.3), volmdlr.Point2D(2, 2))
+        control_points, knots, knot_multiplicities, weights = vme.BSplineCurve.arc_to_nurbs(arc)
+
+        expected_ctrlpts = [
+            [0.0, 0.3], [0.4578268848009306, -0.4077103355939562],
+            [1.2927631248147025, -0.2922077353431394], [2.127699364828474, -0.17670513509232277],
+            [2.3762039959055192, 0.6287170050840547], [2.6247086269825646, 1.4341391452604322],
+            [2.0000000000000004, 1.9999999999999996]
+        ]
+        expected_knots = [0.0, 0.3333333333333333, 0.6666666666666666, 1.0, 1.0]
```

Comment:
To be in line with our definition, it is expected that nodes do not have repeating values. For this case, I think it should be rewritten like this:
       expected_knots = [0.0, 0.3333333333333333, 0.6666666666666666, 1.0]
       expected_knot_multiplicities = [3, 2, 2, 5]
 
 The resulting knot-vector would be U = [0.0, 0.0, 0.0, 0.333, 0.333, 0.6666, 0.6666, 1.0, 1.0, 1.0, 1.0, 1.0]
 
 Verify that this is as expected by the algorithm. By the definition the size of the knot vector should be equal to: len(control_points) + degree + 1. As you have 7 control_points, and I imagine the degree is 2, we should expect a knot vector of size 10. So I think maybe the knot 1.0, with multiplicity 2 should not exist for this case? Can you validate this please?
       
![image](https://github.com/Dessia-tech/volmdlr/assets/89979482/2da52b40-0504-4669-aa5b-d28f4b82ad59)
![image](https://github.com/Dessia-tech/volmdlr/assets/89979482/418efb24-f27f-41b0-82c4-cb9a02c01bba)